components/cdc/src/delegate.rs

```diff
@@ -1053,9 +1050,11 @@ impl Delegate {
     fn sink_delete(&mut self, mut delete: DeleteRequest, rows: &mut RowsBuilder) -> Result<()> {
         match delete.cf.as_str() {
             "lock" => {
-                if self.pop_lock(Key::from_encoded_slice(&delete.key))? != 0 {
+                if ApiV2::parse_key_mode(delete.get_key()) != KeyMode::Raw {
```

Comment:
We don't need this check. RawKV (API v2, and only API v2 can use CDC) has no lock and will write to default cf only.